scripts/utils.js

```diff
@@ -288,21 +320,50 @@ function buildTagsBlock() {
       recBlock.before(tagsBlock);
     }
   } else {
-    main.lastElementChild.append(tagsBlock);
+    mainEl.lastElementChild.append(tagsBlock);
   }
 }
 
+async function buildArticleMeta(mainEl) {
+  const miloLibs = getLibs();
+  const { getMetadata, getConfig } = await import(`${miloLibs}/utils/utils.js`);
+
+  // author + publication date
+  const author = getMetadata('author') || 'Adobe Communications Team';
+  const { codeRoot } = getConfig();
+  const authorURL = getMetadata('author-url') || (author ? `${codeRoot}/authors/${author.replace(/[^0-9a-z]/gi, '-').toLowerCase()}` : null);
+
+  const publicationDate = getMetadata('publication-date');
+
+  const articleMeta = buildBlock('article-meta', [
+    [`<p>${authorURL ? `<a href="${authorURL}">${author}</a>` : author}</p>
+      <p>${publicationDate}</p>`],
+  ]);
+
+  // put article meta (author + link sharings) in front of first content div
+  const allContentDivs = mainEl.querySelectorAll(':scope > div');
+  if (!allContentDivs) return;
+
+  const contentBlock = [...allContentDivs].find((div) => !div.querySelector('.article-hero-banner') && div.querySelector('p'));
+  if (!contentBlock) return;
+
+  contentBlock.insertBefore(articleMeta, contentBlock.firstChild);
+}
+
 export async function buildAutoBlocks() {
   const miloLibs = getLibs();
   const { getMetadata } = await import(`${miloLibs}/utils/utils.js`);
   const mainEl = document.querySelector('main');
+
   try {
     if (getMetadata('content-type') === 'article' && !mainEl.querySelector('.article-header')) {
-      await buildArticleHeader(mainEl);
-      buildTagsBlock();
+      await buildArticleHeroBanner(mainEl);
+      buildTagsBlock(mainEl);
+      await buildArticleMeta(mainEl);
```

Comment:
I'm not sure but it looks like these 3 function calls are not dependant on each other?  If that is the case then `buildArticleHeroBanner` and `buildArticleMeta` can be parallelized